lib/membrane/video_compositor/wgpu/native.ex

```diff
@@ -4,57 +4,38 @@ defmodule Membrane.VideoCompositor.Wgpu.Native do
 
   use Rustler,
     otp_app: :membrane_video_compositor_plugin,
-    crate: "membrane_videocompositor"
+    crate: "membrane_video_compositor"
 
-  alias Membrane.VideoCompositor.VideoTransformations
-  alias Membrane.VideoCompositor.RustStructs.{BaseVideoPlacement, RawVideo}
-
-  @type wgpu_state() :: any()
   @type error() :: any()
-  @type id() :: non_neg_integer()
-  @type frame_with_pts() :: {binary(), Membrane.Time.t()}
-
-  @spec init(RawVideo.t()) :: {:ok, wgpu_state()} | {:error, error()}
-  def init(_out_video), do: error()
-
-  @spec process_frame(wgpu_state(), id(), binary(), Membrane.Time.t()) ::
-          :ok | {:ok, frame_with_pts()} | {:error, atom()}
-  def process_frame(_state, _id, _frame, _pts), do: error()
-
-  @spec force_render(wgpu_state()) :: {:ok, frame_with_pts()} | {:error, atom()}
-  def force_render(_state), do: error()
-
-  @spec add_video(
-          wgpu_state(),
-          id(),
-          RawVideo.t(),
-          BaseVideoPlacement.t(),
-          VideoTransformations.t()
-        ) ::
-          :ok | {:error, error()}
-  def add_video(_state, _id, _stream_format, _placement, _transformations), do: error()
 
-  @spec update_stream_format(wgpu_state(), id(), RawVideo.t()) :: :ok | {:error, error()}
-  def update_stream_format(_state, _id, _stream_format), do: error()
+  @opaque native_state() :: reference()
+  @opaque wgpu_ctx() :: non_neg_integer()
 
-  @spec update_placement(wgpu_state(), id(), BaseVideoPlacement.t()) ::
+  @spec test_scene_deserialization(Membrane.VideoCompositor.Scene.RustlerFriendly.t()) ::
           :ok | {:error, error()}
-  def update_placement(_state, _id, _placement), do: error()
+  def test_scene_deserialization(_scene), do: error()
 
-  @spec update_transformations(wgpu_state(), id(), VideoTransformations.t()) ::
-          :ok | {:error, error()}
-  def update_transformations(_state, _id, _transformations), do: error()
+  @spec init() :: {:ok, native_state()} | {:error, error()}
+  def init(), do: error()
 
-  @spec remove_video(wgpu_state(), id()) ::
-          :ok | {:error, error()}
-  def remove_video(_state, _id), do: error()
+  @spec wgpu_ctx(native_state()) :: wgpu_ctx()
```

Comment:
Don't we want to add some alternative return type for indicating failure?

Reply:
This can't fail

Reply:
This only increments the refcount of a shared pointer and gives you the pointer.